### ISSUE TITLE
Click event when clicking single hit

### ISSUE THREAD

**Issue by ericwallen** [algolia/react-instantsearch#2998]
Hello, I apologize in advance if this is covered somewhere. I can't seem to find an answer to it. 

I need to be able to know when a hit is clicked. I put my search inside of a modal and need to hide the modal whenever someone clicks on a result. Is this possible?

Thank you!

**Comment by Haroenv** [comment]
There's a couple of options, depending on how you made your search:

- you can add an event listener to the container of the hits, since click events bubble up
- you likely hide the modal because you set some state around the result, could you close the modal there?
- adding an event listener to the wrapper for a single hit is also possible in some cases
- you can use `connectHits` to make a fully custom hits, with event listeners on the hit containers

**Comment by ericwallen** [comment]
Haroenv, you're the best! `connectHits` is exactly what I needed. Using it I was able to call the function that changed the state of the parent component which is where the hiding and showing of the modal logic are located. Obviously, I had to pass the function through the props of every component. I only mention that for anyone reading this in the future.

I really appreciate it. 

Sincerely,
Eric
